filters/auth/jwt_validation_test.go

```diff
@@ -0,0 +1,210 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/net"
+	"github.com/zalando/skipper/proxy/proxytest"
+)
+
+const (
+	kid = "mykid"
+)
+
+var (
+	privateKey, _ = rsa.GenerateKey(rand.Reader, 2048)
+)
+
+func createToken() string {
+	// Create the Claims
+	claims := &jwt.StandardClaims{
+		ExpiresAt: time.Now().Unix() + 1000,
+		Issuer:    "test",
+		Subject:   "aaa",
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+	token.Header["kid"] = kid
+
+	s, err := token.SignedString(privateKey)
+
+	fmt.Printf("Token: %v %v", s, err)
+
+	return s
+}
+
+func TestToken(t *testing.T) {
+	s := createToken()
+
+	publicKeyString := base64.RawURLEncoding.EncodeToString(privateKey.PublicKey.N.Bytes())
+
+	parsedToken, err := jwt.Parse(s, func(token *jwt.Token) (interface{}, error) {
+		return parsePublicKey(publicKeyString), nil
+	})
+
+	if err != nil {
+		t.Errorf("Failed to json decode: %v", err)
+		return
+	}
+
+	err = parsedToken.Claims.Valid()
+
+	if err != nil {
+		t.Errorf("Failed token: %v", err)
+		return
+	}
+
+}
+
+func parsePublicKey(encodedKey string) *rsa.PublicKey {
```

Comment:
I think there is a method to get public key from private key https://pkg.go.dev/crypto/rsa#PrivateKey.Public